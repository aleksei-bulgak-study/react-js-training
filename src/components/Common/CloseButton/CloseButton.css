.close-button {
  background-color: transparent;
  color: transparent;
  border: transparent;
  outline: none;
}

.close-button::before {
  content: '';
  position: absolute;
  border-top: 2px solid var(--white);
  transform: rotate(45deg);
  left: 0%;
  top: 50%;
}

.close-button::after {
  content: '';
  position: absolute;
  border-top: 2px solid var(--white);
  transform: rotate(-45deg);
  left: 0%;
  top: 50%;
}

.close-button.small {
  width: 1rem;
  height: 1rem;
}

.close-button.small::after,
.close-button.small::before{
  width: 1rem;
}

.close-button.large {
  width: 2rem;
  height: 2rem;
}

.close-button.large::after,
.close-button.large::before{
  width: 2rem;
}
