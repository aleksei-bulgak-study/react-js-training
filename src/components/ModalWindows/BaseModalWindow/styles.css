.modal-window__overlay {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: var(--grey-darker);
  filter: blur(1px) grayscale(50%);
  opacity: 0.8;
  z-index: 2;
}

.modal-window {
  background-color: var(--grey-darker);
  color: var(--white);
  position: fixed;
  left: 50vw;
  top: 50vh;
  max-height: 80vh;
  min-width: 500px;
  width: 40%;
  transform: translate(-50%, -50%);
  border-radius: 4px;
  box-shadow: 0 0 10px 0 #000000b3;
  padding: 4rem 5rem;
  overflow: auto;
  z-index: 2;
}

.modal-window__wrapper::after {
  content: '';
  clear: both;
  display: block;
}

.modal-window__title {
  text-transform: uppercase;
  font-weight: 300;
  font-size: 3rem;
  line-height: 3rem;
  letter-spacing: 0.2rem;
  margin-bottom: 1rem;
}

.modal-window__close {
  position: absolute;
  right: 2rem;
  top: 2rem;
}

@media screen and (max-width: 900px) {
  .modal-window {
    width: 100%;
    height: 100%;
    max-height: 100%;
    padding: 2rem 2rem;
    z-index: 5;
  }
}
