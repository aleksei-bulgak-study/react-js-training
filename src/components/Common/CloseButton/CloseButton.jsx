import React from 'react';
import PropTypes from 'prop-types';
import './CloseButton.css';

const SIZES = ['small', 'large'];

const CloseButton = ({ size, className, onClick }) => (
  <button
    type="button"
    className={`close-button ${className} ${size}`}
    onClick={onClick}
  >
    Close
  </button>
);

CloseButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  size: PropTypes.oneOf(SIZES),
  className: PropTypes.string,
};

CloseButton.defaultProps = {
  size: SIZES[1],
  className: '',
};

export default CloseButton;
