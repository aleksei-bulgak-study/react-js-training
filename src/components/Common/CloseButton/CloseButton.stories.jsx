/* eslint-disable react/prop-types */
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { withKnobs, select } from '@storybook/addon-knobs';
import CloseButton from './CloseButton';
import '../../../app.css';

export default {
  title: 'CloseButton',
  decorators: [withKnobs],
};

const options = {
  small: 'small',
  large: 'large'
};

export const Default = () => (
  <CloseButton
    // eslint-disable-next-line no-alert
    onClick={() => alert('test message')}
    size={select('Size', options, options.small)}
  />
);
