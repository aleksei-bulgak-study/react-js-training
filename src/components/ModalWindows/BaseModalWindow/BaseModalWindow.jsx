import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';
import CloseButton from '../../Common/CloseButton';

const BaseModalWindow = ({ title, onClose, children, className, Logo }) => (
  <>
    <div className="modal-window__overlay" />
    <div className="modal-window">
      <div className={`modal-window__wrapper ${className}`}>
        {Logo}
        <h1 className="modal-window__title">{title}</h1>
        <CloseButton
          onClick={onClose}
          size="large"
          className="modal-window__close"
        />
        {children}
      </div>
    </div>
  </>
);

BaseModalWindow.propTypes = {
  title: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  className: PropTypes.string,
  Logo: PropTypes.node,
};

BaseModalWindow.defaultProps = {
  className: '',
  Logo: null,
};

export default BaseModalWindow;
