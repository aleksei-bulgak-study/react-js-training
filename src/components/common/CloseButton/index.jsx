import React from 'react';
import PropTypes from 'prop-types';
import './styles.css';

const SIZES = ['small', 'large'];

const CloseButton = ({ size, positonStyles, onClick }) => (
  <button
    type="button"
    className={`close-button ${positonStyles} ${size}`}
    onClick={onClick}
  >
    Close
  </button>
);

CloseButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  size: PropTypes.oneOf(SIZES),
  positonStyles: PropTypes.string,
};

CloseButton.defaultProps = {
  size: SIZES[1],
  positonStyles: '',
};

export default CloseButton;
