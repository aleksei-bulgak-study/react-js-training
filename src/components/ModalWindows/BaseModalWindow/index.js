import BaseModalWindow from './BaseModalWindow';

export default BaseModalWindow;
