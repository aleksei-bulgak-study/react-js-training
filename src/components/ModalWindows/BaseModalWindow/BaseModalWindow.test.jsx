import React from 'react';
import { shallow } from 'enzyme';
import BaseModalWindow from './BaseModalWindow';
import CloseButton from '../../Common/CloseButton';

describe('BaseModalWindow', () => {
  test('make snapshot', () => {
    const Logo = "logo component";
    const onClick = jest.fn();
    const modal = shallow(
      <BaseModalWindow
        title="test title"
        onClose={onClick}
        className="additionalClass"
        Logo={Logo}
      >
        internal data component
      </BaseModalWindow>,
    );

    expect(modal.html()).toMatchSnapshot();
    modal.find(CloseButton).simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
