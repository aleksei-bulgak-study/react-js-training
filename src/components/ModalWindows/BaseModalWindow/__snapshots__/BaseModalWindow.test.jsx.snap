// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseModalWindow make snapshot 1`] = `"<div class=\\"modal-window__overlay\\"></div><div class=\\"modal-window\\"><div class=\\"modal-window__wrapper additionalClass\\">logo component<h1 class=\\"modal-window__title\\">test title</h1><button type=\\"button\\" class=\\"close-button modal-window__close large\\">Close</button>internal data component</div></div>"`;
